### PR TITLE
Ns/bench zk cs ghl

### PR DESCRIPTION
**Pull request by nsarlin-zama**
### PR content/description
Adds a bench to compare ghl and cs bounds for zk v2. Also add some limit tests for zk with extreme but supported values

### CODE REVIEW

**Comment by nsarlin-zama** [comment]
manual run of the bench workflow: https://github.com/zama-ai/tfhe-rs/actions/runs/12033323684/job/33547332621

File: tfhe-zk-pok/src/proofs/pke_v2.rs [hunk 1]
```diff
@@ -499,10 +499,15 @@ pub fn compute_crs_params(
...
+            // Result is multiplied and divided by 10000 to avoid floating point operations.
+            // This could be avoided if one day we need to support bigger params.
             Bound::GHL => 950625,
-            Bound::CS => (2 * (d + k) + 4) as u128,
-        }) * (B_squared + (sqr(d + 2) * (d + k)) as u128 / 4)
+            Bound::CS => 2 * (d as u128 + k as u128) + 4,
+        })
+        .checked_mul(B_squared + (sqr((d + 2) as u128) * (d + k) as u128) / 4)
+        .unwrap_or_else(|| {
+            panic!("Invalid parameters for zk_pok, B_squared: {B_squared}, d: {d}, k: {k}")
```
Comment (IceTDrinker) [inline tfhe-zk-pok/src/proofs/pke_v2.rs:509]: a clearer error message please

File: tfhe-zk-pok/src/proofs/pke_v2.rs [hunk 2]
```diff
@@ -513,6 +518,13 @@ pub fn compute_crs_params(
     // Since we use B_bound_square, the log is divided by 2
     let m_bound = 1 + ceil_ilog2(B_bound_squared).div_ceil(2) as usize;
 
+    // m_bound is used to do the bit decomposition of a u64 integer, so we check that it can be
+    // safely used for this
+    assert!(
+        m_bound <= 64,
+        "Invalid parameters for zk_pok, B_squared: {B_squared}, d: {d}, k: {k}, m_bound: {m_bound} > 64"
```
Comment (IceTDrinker) [inline tfhe-zk-pok/src/proofs/pke_v2.rs:525]: same here for the m_bound could be "we only support 64 bits integer, the computed m parameter is {m_bound} bits > 64 bits"

File: tfhe-zk-pok/benches/pke_v2.rs [hunk 3]
```diff
@@ -16,35 +18,36 @@ fn bench_pke_v2_prove(c: &mut Criterion) {
...
         let effective_t = params.t >> 1;
         let bits = (params.k as u32) * effective_t.ilog2();
 
-        for load in [ComputeLoad::Proof, ComputeLoad::Verify] {
-            let zk_load = match load {
-                ComputeLoad::Proof => "compute_load_proof",
-                ComputeLoad::Verify => "compute_load_verify",
-            };
-            let bench_id = format!("{bench_name}::{param_name}_{bits}_bits_packed_{zk_load}");
+        let bench_id = format!("{bench_name}::{param_name}_{bits}_bits_packed_{load}_{bound:?}");
```
[the remark this answers is not in the document]
Comment (nsarlin-zama) [reply tfhe-zk-pok/benches/pke_v2.rs:36]: It will print "GHL" or "CS", which is ok as David confirmed

**Review by IceTDrinker — APPROVED** [review]

**Pull request merged**